lib/Dialect/FIRRTL/Transforms/InferResets.cpp

```diff
@@ -1714,14 +1714,17 @@ LogicalResult InferResetsPass::implementAsyncReset(FModuleOp module,
       if (nodeOp && !dom.dominates(nodeOp.getInput(), opsToUpdate[0])) {
         LLVM_DEBUG(llvm::dbgs()
                    << "- Promoting node to wire for move: " << nodeOp << "\n");
-        ImplicitLocOpBuilder builder(nodeOp.getLoc(), nodeOp);
+        auto builder = ImplicitLocOpBuilder::atBlockBegin(nodeOp.getLoc(),
+                                                          nodeOp->getBlock());
         auto wireOp = builder.create<WireOp>(
             nodeOp.getResult().getType(), nodeOp.getNameAttr(),
             nodeOp.getNameKindAttr(), nodeOp.getAnnotationsAttr(),
             nodeOp.getInnerSymAttr(), nodeOp.getForceableAttr());
-        emitConnect(builder, wireOp.getResult(), nodeOp.getInput());
+        // Don't delete the node, since it might be in use in worklists.
         nodeOp->replaceAllUsesWith(wireOp);
-        nodeOp.erase();
+        nodeOp->removeAttr(nodeOp.getInnerSymAttrName());
```

Comment:
Should probably remove the annotations as well, e.g. if the node has an OMIR path tracker we will end up having two things with the same path tracker.

Not as important but might want to make the Node non-forcible.

Reply:
yea working on it.

Reply:
Changing the node to be non-forcable is kind of a pain as it would requiring changing the number of results, which I don't think you can do.  The forcable result should be unused, so it should be cleared away with the rest of the node by CSE/Canonicalization.

Reply:
Yes you can't change results, the way it's done is a bit rough but basically creates a new one and replaces it.

And even if you did that (there's a helper, but in the `detail` namespace) it'd be erasing the existing node so no-go.

Thanks!!